include/dfrn.php

```diff
@@ -817,46 +877,61 @@ private function entry($doc, $type, $item, $owner, $comment = false, $cid = 0) {
 
 		xml::add_element($doc, $entry, "activity:verb", construct_verb($item));
 
-		if ($item['object-type'] != "")
+		if ($item['object-type'] != "") {
 			xml::add_element($doc, $entry, "activity:object-type", $item['object-type']);
-		elseif ($item['id'] == $item['parent'])
+		} elseif ($item['id'] == $item['parent']) {
 			xml::add_element($doc, $entry, "activity:object-type", ACTIVITY_OBJ_NOTE);
-		else
+		} else {
 			xml::add_element($doc, $entry, "activity:object-type", ACTIVITY_OBJ_COMMENT);
+		}
 
 		$actobj = self::create_activity($doc, "activity:object", $item['object']);
-		if ($actobj)
+		if ($actobj) {
 			$entry->appendChild($actobj);
+		}
 
 		$actarg = self::create_activity($doc, "activity:target", $item['target']);
-		if ($actarg)
+		if ($actarg) {
 			$entry->appendChild($actarg);
+		}
 
 		$tags = item_getfeedtags($item);
 
-		if(count($tags)) {
-			foreach($tags as $t)
-				if (($type != 'html') OR ($t[0] != "@"))
+		if (count($tags)) {
+			foreach ($tags as $t) {
+				if (($type != 'html') OR ($t[0] != "@")) {
 					xml::add_element($doc, $entry, "category", "", array("scheme" => "X-DFRN:".$t[0].":".$t[1], "term" => $t[2]));
+				}
+			}
 		}
 
-		if(count($tags))
-			foreach($tags as $t)
-				if ($t[0] == "@")
+		if (count($tags)) {
+			foreach ($tags as $t) {
+				if ($t[0] == "@") {
 					$mentioned[$t[1]] = $t[1];
+				}
+			}
+		}
 
 		foreach ($mentioned AS $mention) {
 			$r = q("SELECT `forum`, `prv` FROM `contact` WHERE `uid` = %d AND `nurl` = '%s'",
 				intval($owner["uid"]),
 				dbesc(normalise_link($mention)));
-			if ($r[0]["forum"] OR $r[0]["prv"])
+
+			if (!dbm::is_result($r)) {
```

Comment:
`dbm::is_result()` is for determining if a result has been returned, not if rows has (>=1 count) has been returned. Maybe I need to introduce another method, e.g. `dbm::has_rows()` (I prefer camel-case for method names here but I go with this project's style rule).